scripts/directtools/__init__.py

```diff
@@ -0,0 +1,450 @@
+from scipy import constants
+import time
+
+
+def _chooseMarker(markers, index):
+    """Pick a marker from markers and return next cyclic index."""
+    if index >= len(markers):
+        index = 0
+    marker = markers[index]
+    index += 1
+    return marker, index
+
+
+def _clearlatex(s):
+    """Return a string from which LaTeX formatting has been removed."""
+    s = s.translate(None, '$\\^{}')
+    s = s.replace('$', '')
+    s = s.replace('\\', '')
+    s = s.replace('^', '')
+    s = s.replace('{', '')
+    s = s.replace('}', '')
+    return s
+
+
+def _finalizeprofileE(axes):
+    """Set axes for const E axes."""
+    axes.set_xlim(xmin=0.)
+    axes.set_xlabel('$Q$ (\\AA$^{-1}$)')
+    axes.set_ylim(0.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto Q-range: {}...{} \xc5-1'.format(xMin, xMax))
+
+
+def _finalizeprofileQ(workspaces, axes):
+    """Set axes for const Q axes."""
+    workspaces = _normwslist(workspaces)
+    axes.set_xlim(xmin=-10.)
+    axes.set_xlabel('Energy (meV)')
+    cMax = 0.
+    for i in range(len(workspaces)):
+        c = numpy.nanmax(workspaces[0].readY(0))
+        if c > cMax:
+            cMax = c
+    axes.set_ylim(ymin=0., ymax=cMax / 100.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto E-range: {}...{} meV'.format(xMin, xMax))
+
+
+def _globalnanminmax(workspaces):
+    """Return global minimum and maximum of the given workspaces."""
+    workspaces = _normwslist(workspaces)
+    globalMin = numpy.inf
+    globalMax = -numpy.inf
+    for ws in workspaces:
+        cMin, cMax = nanminmax(ws)
+        if cMin < globalMin:
+            globalMin = cMin
+        if cMax > globalMax:
+            globalMax = cMax
+    return globalMin, globalMax
+
+
+def _label(ws, cut, width, singleWS, singleCut, singleWidth, quantity, units):
+    """Return a line label for a line profile."""
+    ws = _normws(ws)
+    logs = SampleLogs(ws)
+    wsLabel = ''
+    if not singleWS:
+        logs = SampleLogs(ws)
+        T = numpy.mean(logs.sample.temperature)
+        wsLabel = '\\#{:06d} $T$ = {:0.1f} K $E_i$ = {:0.2f} meV'.format(logs.run_number, T, logs.Ei)
+    cutLabel = ''
+    if not singleCut or not singleWidth:
+        cutLabel = quantity + ' = {:0.2f} $\pm$ {:1.2f}'.format(cut, width) + ' ' + units
+    return wsLabel + ' ' + cutLabel
+
+
+def _normws(workspace):
+    """Retrieve workspace from mtd if it is a string, otherwise return as-is."""
+    name = str(workspace)
+    if name:
+        workspace = mtd[name]
+    return workspace
+
+
+def _normwslist(workspaces):
+    """Retrieve workspaces from mtd if they are string, otherwise return as-is."""
+    if not isinstance(workspaces, collections.Iterable) or isinstance(workspaces, str):
+        workspaces = [workspaces]
+    unnormWss = workspaces
+    return [_normws(ws) for ws in unnormWss]
+
+
+def _plottingtime():
+    """Return a string presenting the plotting time."""
+    return time.strftime('%d.%m.%Y %H:%M:%S')
+
+
+def _points(edges):
```

Comment:
`_points` doesn't describe what this does very well - could it be called something like `_binCentres` instead?

Reply:
The rough equivalent in C++ is `MatrixWorkspace::points()`, but I think you're right. `_binCentres` would describe the action better.